robot/lib/Inc/drivers/Battery.hpp

```diff
@@ -1,28 +1,47 @@
 #pragma once
 
 #include "cstdint"
-
+/** @class Battery
+ * Interfaces with analog pin to get voltage-related data
+ *
+ * @note As of May 2020, this class does not currently work, as it does not read analog pin
+ */
 class Battery {
 public:
     Battery();
 
-    void update();
+    void update();             /**< Updates reading from analog pin*/
 
-    float getBattPercentage();
+    float getBattPercentage(); /**< Returns the last battery percentage read */
 
-    uint8_t getRaw();
+    uint8_t getRaw();          /**< Returns battery voltage as a raw 8-bit integer */
 
-    bool isBattCritical();
+    bool isBattCritical();     /**< Stores whether battery is critical (at or below 0%) */
 
 private:
-    // 5-cell lipo * 4.3 v per cell = 21.5 battery volts
-    // 68k and 10k ohm voltage divider, analog in read votlage 2.756
+    /**
+     * The maximum voltage the battery can safely hold
+     *
+     * Calculations:
+     *  - 5-cell lipo * 4.3 v per cell = 21.5 battery volts
+     *  - 68k and 10k ohm voltage divider, analog in read voltage 2.756
+     *
+     *  @note Voltages above this threshold indicate that the battery is overvolted!
+     */
     const float MAX_SAFE_BATT_VOLTAGE_READ = 2.756;
 
-    // 5-cell lipo * 3.0 v per cell = 15.0 battery volts
-    // 68k and 10k ohm voltage divider, analog in read votlage 1.923
+    /**
+    * The minimum voltage the battery can safely hold
+    *
+    * Calculations:
+    *  - 5-cell lipo * 4.3 v per cell = 21.5 battery volts
+    *  - 68k and 10k ohm voltage divider, analog in read voltage 1.923
+    */
     const float MIN_SAFE_BATT_VOLTAGE_READ = 1.923;
```

Comment:
Is this not dependent on our current battery, which is changing?

Reply:
@mmaisonneuve3 can you confirm the batteries we are ordering and what their voltages are?